### PR TITLE
Updating header and footer UI

### PR DESCRIPTION
**Pull request by marlincommuincations**
Tasks completed within PR: 
- updating the footer column widths 
- updating the header to be fixed and transparent/white background along with logo resize + removing unwanted scss files

In order to preview the latest updates that were made to the header and footer, the CHILD theme code needs to be updated. The following steps can be followed in order to update your local environment: 

- Navigate to the CHILD theme folder within the planet-4 development setup (e.g. `cd planet4/themes/planet4-child-theme-australiapacific`)
- Once within the CHILD theme run the command `git fetch && git checkout feature/header-footer-ui-updates`
- The above command will checkout the branch and will update your local environment which will include the latest footer and header setup
- The header and footer can be visually tested by navigating to the homepage of your WordPress setup. (e.g. http://www.planet4.test/)

Header Transparent:
image

Header scrolled:
image

Footer:
image

### CODE REVIEW

**Review by techieshark — CHANGES_REQUESTED** [review]
Hi @marlincommuincations thanks for this! Would you mind putting some "How to test" instructions in the PR body - maybe some commands to run so Kara et al can see this locally in their dev environment? Screenshots would be great too! Thanks

**Review by marlincommuincations — COMMENTED** [review]
PR comment updated.

**Comment by techieshark** [comment]
@marlincommuincations thanks for updating that PR! I believe @ShoryuDZ and @karabishopgreenpeace will be reviewing this later today!

**Review by ShoryuDZ — CHANGES_REQUESTED** [review]
I haven't been able to sync content (from ElastiPress?) into the page so I don't have the same content as you at the moment, however I'm still able to see how your changes look over some dummy pages (even if the dummy pages have images in the background which make it difficult to see the menu items when transparent, I can still get a feel for how it is supposed to work).
The main issue I'm seeing at the moment is that when the menu is transparent, the menu items turn the same colour as the background (i.e. White) - making it impossible to see:
Screenshot 2023-12-14 at 6 47 23 pm
This doesn't appear the be an issue when the background of the menu is solid white:
Screenshot 2023-12-14 at 6 47 34 pm
Let me know if you're able to see/resolve this issue as well, or if you think there's something wrong with my setup.

**Comment by karabishopgreenpeace** [comment]
Just an FYI @Shoryu Das-Zaman ***@***.***> , the menu
reversed to white is only meant to be on the home page.

**Comment by ShoryuDZ** [comment]
@karabishopgreenpeace Thanks for letting me know, I will check to see that effect is only happening on the Home page once code has been resubmitted - currently, it is happening on every single page as far as I'm aware

**Comment by marlincommuincations** [comment]
@ShoryuDZ @techieshark I have gone ahead and updated the menu reserve to only occur on the homepage. All other pages will have the default white background header.

**Review by ShoryuDZ — APPROVED** [review]
Happy with how this looks, thanks @marlincommuincations!

File: package-lock.json
Comment (techieshark) [inline package-lock.json:1]: I don't think package-lock.json needs to change for us to get the UI changes we're after in this PR right? Can this be removed from this PR?

**Review by techieshark — CHANGES_REQUESTED** [review]
Good morning @marlincommuincations thanks for your extensive work on this! 

I finally had some time to review the code, please see comments in PR. Thank you.

---

Update - comments from myself, Kara & Bianca [in Slack](https://gpap.slack.com/archives/C05QXT9P414/p1705870800686859):

> it feels like the grow/shrink effect is too fast. Ideally the speed matches the speed you are scrolling or something. If I’m slowly scrolling, the shrink should be slow and almost not noticed

> [Re: the logo size transition] Slower would be much nicer if you can change the transition speed. Ideally the white panel would have a transition as well (maybe a fade or slide down if it's easy enough).

Also the alignment / margin on the logo seems to change, so it almost bounces left to right; the left alignment should not change on animation; see this page for example https://donatehubspot.greenpeace.org.au/test-header

File: style/header.scss [hunk 1]
```diff
@@ -1,15 +1,82 @@
...
+
+  #nav-mobile {
     display: none;
   }
+
+  .site-logo {
+    transition: all 0.3s;
+
+    &.largelogo {
+      width: 260px;
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply style/header.scss:49]: @techieshark These values were provided with the logo resize code solution. These were provided by Kara as this solution was built internally and not within Marlin's scope.

File: style/global.scss [hunk 2]
```diff
@@ -21,14 +22,30 @@ h6{
...
   .container-xxl {
-    &:not(.narrow-container){
+    &:not(.narrow-container) {
       max-width: 1600px;
     }
   }
+}
+
+.page-content {
+  padding-top: 0 !important;
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply style/global.scss:39]: Will remove the !important declaration

File: js/scripts.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+window.onscroll = function () {
+	growShrinkLogo();
+};
+
+let size = 0;
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply js/scripts.js]: @techieshark I have updated the variable name

File: js/scripts.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  const logo = document.getElementById('logo');
+  const header = document.getElementById('header');
+  const endOfDocumentTop = 1;
+  const scroll =
+		window.pageYOffset ||
+		document.documentElement.scrollTop ||
+		document.body.scrollTop ||
+		0;
+
+	if (document.body.classList.contains('home')) {
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply js/scripts.js]: @techieshark I have updated the functions.php file to only load the JS file if its the front page.

File: js/scripts.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+		if (size == 0 && scroll > endOfDocumentTop) {
+			logo.classList.remove( 'largelogo' );
+			logo.classList.add('smalllogo');
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply js/scripts.js]: Spacing has been resolved.

File: js/scripts.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+window.onscroll = function () {
+	growShrinkLogo();
+};
+
+let size = 0;
+
+function growShrinkLogo() {
+  const logo = document.getElementById('logo');
+  const header = document.getElementById('header');
+  const endOfDocumentTop = 1;
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply js/scripts.js]: I have renamed the variable to reflect the requirement.

File: functions.php [hunk 7]
```diff
@@ -15,32 +15,7 @@ function enqueue_child_styles() {
...
-
-function fix_svg() {
-  echo '<style type="text/css">
-        .attachment-266x266, .thumbnail img {
-             width: 100% !important;
-             height: auto !important;
-        }
-        </style>';
+function register_custom_script() {
+  wp_enqueue_script( 'marlin-script', get_stylesheet_directory_uri() .  '/js/scripts.js' );
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply functions.php]: Script name and handle updated.

File: style/global.scss [hunk 8]
```diff
@@ -21,14 +22,30 @@ h6{
...
 @media (min-width: 1600px) {
+
   .container,
   .container-lg,
   .container-md,
   .container-sm,
   .container-xl,
   .container-xxl {
-    &:not(.narrow-container){
+    &:not(.narrow-container) {
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply style/global.scss:32]: I have added in the spacing as a different commit.

File: style.css
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply style.css:1]: All the styles in the style.css file are auto generated. Not sure I understand the requirement here.

File: js/scripts.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+window.onscroll = function () {
+	growShrinkLogo();
```
[the remark this answers is not in the document]
Comment (marlincommuincations) [reply js/scripts.js]: @techieshark Happy to update the solution, however this was code provided by Kara as the header modifications are out of scope for Marlin. If the updated solution can be provided I am happy to implement it into the child theme.

File: package-lock.json
Comment (marlincommuincations) [reply package-lock.json:1]: I have removed the package-lock.json file updates.

File: js/scripts.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (techieshark) [reply js/scripts.js]: Thanks!

**Review by techieshark — COMMENTED** [review]
hi Adrian! @marlincommuincations thanks very much for addressing some of those requests from our end.

I'm interested in having a pretty clean git commit history (so us or future devs can look back and make sense of everything). 

With that in mind, I wonder if I could trouble you to do the following:

* first make a PR which copies in any files from the master theme you need to alter
* then make a new PR just for the footer bit
* and a different PR for the header changes
* and any changes to the generated style.css file could be in their own git commit with a message header like "Generated style.css" and commit message indicating what command was used to do that conversion from sass to css. 

Thanks!

**Comment by techieshark** [comment]
Closing as this has been actioned in separate PRs now - thank you!

**Pull request closed without merging**